File: python/백준/그리디/2812 크게 만들기.py
N, K = map(int, input().split())

num = input()

stack = []
answer = []
count = 0
for digit in num:

    while stack and stack[-1] < int(digit) and count < K:
        stack.pop()
        count += 1

    stack.append(int(digit))

if count < K:
    for _ in range(K-count):
        stack.pop()

for i in range(len(stack)):
    print(stack[i], end="")

File: cpp/백준/그리디/2812 크게 만들기.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int N, K;
    cin >> N >> K;

    string num;
    cin >> num;

    vector<char> stack;
    int count = 0;
    for (char digit : num) {
        while (!stack.empty() && stack.back() < digit && count < K) {
            stack.pop_back();
            count += 1;
        }
        stack.push_back(digit);
    }

    if (count < K) {
        for (int i = 0; i < K - count; ++i) {
            stack.pop_back();
        }
    }

    for (char ch : stack) cout << ch;
    cout << "\n";
    return 0;
}
